src/tests/mod.rs

```diff
@@ -16,7 +16,8 @@ mod trie_tests {
         let root_hash = trie.root_hash().unwrap();
         let rs = format!("0x{}", hex::encode(root_hash.clone()));
         assert_eq!(rs.as_str(), hash);
-        let mut trie = EthTrie::from(Arc::clone(&memdb), root_hash).unwrap();
+
+        let mut trie = trie.at_root(root_hash);
```

Comment:
*nitpick* just the naming convention now is a bit unclear. i.e. you can create multiple `trie`s from a single `trie` based on different state roots. Something like `let mut trie_db = EthTrieDB::new(..)` or `let mut trie_store = EthTrieStore(..)`. But, tbh it's just a minor nitpick, i'm not convinced it's worth the changing the whole api

Reply:
Wait now _I'm_ confused. I don't follow what you're saying about `trie_db` or `trie_store`. This is still a regular old `EthTrie` object that gets returned. Maybe let's do a short call on Monday, because I have a feeling this is something straightforward, but might be slow to figure out over text.